zeebe/engine/src/main/java/io/camunda/zeebe/engine/processing/processinstance/ProcessInstanceModificationModifyProcessor.java

```diff
@@ -194,7 +194,8 @@ public void processRecord(final TypedRecord<ProcessInstanceModificationRecord> c
         new AuthorizationRequest(
                 command,
                 AuthorizationResourceType.PROCESS_DEFINITION,
-                PermissionType.UPDATE_PROCESS_INSTANCE)
+                PermissionType.UPDATE_PROCESS_INSTANCE,
+                value.getTenantId())
```

Comment:
❌ This is wrong. When cancelling a process instance users don't specify a tenant id on the command. Only a process instance key. What we should do here is use the tenant id that's on this process instance:

`processInstance.getValue().getTenantId()`

Reply:
Good catch 👍